code/game/objects/items/weapons/twohanded.dm

```diff
@@ -1127,3 +1128,108 @@
+	name = "supermatter halberd"
+	desc = "The revered weapon of Oblivion Enforcers, used to enforce the Order's will."
+	icon_state = "smhalberd0"
+	force = 5
+	sharp = TRUE
+	damtype = BURN
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = SLOT_BACK
+	throwforce = 15
+	force_unwielded = 5
+	force_wielded = 40
+	toolspeed = 0.25
+	attack_verb = list("enlightened", "enforced", "cleaved", "stabbed", "whacked")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	resistance_flags = FIRE_PROOF
+	var/static/list/obliteration_targets = list(/turf/simulated/wall, /obj/machinery/door/airlock)
+	/// Whether we'll knockdown on hit
+	var/charged = TRUE
+
+/obj/item/twohanded/supermatter/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_FORCES_OPEN_DOORS_ITEM, ROUNDSTART_TRAIT)
+	ADD_TRAIT(src, TRAIT_SUPERMATTER_IMMUNE, ROUNDSTART_TRAIT) //so it can't be dusted by the SM
+	AddComponent(/datum/component/parry, _stamina_constant = 0, _stamina_coefficient = 0.25, _parryable_attack_types = ALL_ATTACK_TYPES)
+
+/obj/item/twohanded/supermatter/update_icon_state()
+	icon_state = "smhalberd[wielded]"
+
+/obj/item/twohanded/supermatter/afterattack(atom/A, mob/user, proximity)
+	if(!proximity)
+		return
+
+	if(!wielded)
+		return
+
+	if(istype(A, /obj/structure/window) || istype(A, /obj/structure/grille)) //same behavior as a fireaxe for windows
+		var/obj/structure/W = A
+		W.obj_destruction("fireaxe")
+
+	//dusting dead people + knocking down people
+	if(isliving(A))
+		var/mob/living/target = A
+		if(target.stat == DEAD)
+			visible_message("<span class='danger'>[user] raises [src] high, ready to bring it down on [target]!</span>")
+			if(do_after(user, 1 SECONDS, TRUE, target))
+				visible_message("<span class='danger'>[user] brings down [src], obliterating [target] with a heavy blow!</span>")
+				playsound(loc, 'sound/effects/supermatter.ogg', 50, TRUE)
+				target.dust()
+				return
+			to_chat(user, "<span class='notice'>You lower [src]. There'll be time to obliterate them later...</span>")
+			return
+
+		if(charged)
+			playsound(loc, 'sound/magic/lightningbolt.ogg', 5, TRUE)
+			target.visible_message("<span class='danger'>[src] flares with energy and shocks [target]!</span>", \
+									"<span class='userdanger'>You're shocked by [src]!</span>", \
+									"<span class='warning'>You hear shocking.</span>")
+			target.KnockDown(4 SECONDS)
+			do_sparks(3, FALSE, src)
+			charged = FALSE
+			addtimer(CALLBACK(src, PROC_REF(recharge)), 4 SECONDS)
+
+	//walls and airlock obliteration logic
+	if(!is_type_in_list(A, obliteration_targets))
+		return
+
+	if(iswallturf(A))
+		if(istype(A, /turf/simulated/wall/indestructible))
+			return
+
+		to_chat(user, "<span class='notice'>You start to obliterate [A].</span>")
+		playsound(loc, hitsound, 50, TRUE)
+
+		var/obj/effect/temp_visual/obliteration_rays/rays = new(get_turf(A))
+
+		if(do_after(user, 5 SECONDS * toolspeed, target = A))
+			new /obj/effect/temp_visual/obliteration(get_turf(A), A)
```

Comment:
Isn't A already a turf here? Do you need to call the `get_turf()`s here? You can probably just make a cast above

Reply:
RCDs use the get_turf() call and that's where I copied this from originally, I'll test if it works with just A and commit it if it does